atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -438,7 +438,10 @@ private NavigableMap<byte[], RowResult<byte[]>> getRowsInternal(
                     SafeArg.of("resultSize", results.size()),
                     SafeArg.of("timeTakenMillis", getRowsMillis));
         }
-        validatePreCommitRequirementsOnReadIfNecessary(tableRef, getStartTimestamp());
+        validatePreCommitRequirementsOnReadIfNecessary(
+                tableRef,
+                getStartTimestamp(),
+                true /* can't skip lock check as we don't know how many cells to expect for the column selection */);
```

Comment:
Question for reviewer: is this true? Or is there actually some way we can know the number of cells to expect? or some other reason why we could skip the check here?